src/magic_folder/test/cli/common.py

```diff
@@ -26,6 +29,66 @@
     MagicFolderCommand,
     run_magic_folder_options,
 )
+from ...api_cli import (
+    MagicFolderApiCommand,
+    run_magic_folder_api_options,
+)
+
+
+
+@contextmanager
+def _pump_client(http_client):
+    """
+    Periodically flush all data in the HTTP client while this context manager
+    is active.
+
+    py:`treq.testing.RequestTraversalAgent` doesn't automatically pass data
+    between a client and server, if a response isn't generated synchronously,
+    so if we want requests to complete, we need to call ``.flush`` on it.
+
+    :param treq.client.HTTPClient http_client:
+        An HTTP Client wrapping a :py:`treq.testing.RequestTraversalAgent`.
+    """
+    def pump():
+        while True:
+            http_client._agent.flush()
+            yield
+
+    coop = Cooperator()
+    task = coop.cooperate(pump())
+    try:
+        yield
+    finally:
+        task.stop()
+        coop.stop()
```

Comment:
Ugh.  It is quite dismaying to see this kind of general infrastructure _still_ being written and rewritten in application-specific test suite helper libraries...

I guess this comment is not actionable.